controller/iteration_blackbox_test.go

```diff
@@ -763,3 +765,208 @@ func assertChildIterationLinking(t *testing.T, target *app.Iteration) {
+			fxt.IterationByName("child 1"),
+			fxt.IterationByName("child 1.2"),
+			fxt.IterationByName("child 1.2.3"),
+			fxt.IterationByName("child 1.2.3.4"),
+		}
+		for _, i := range deletedIterations {
+			_, err := rest.db.Iterations().Load(svc.Context, i.ID)
+			require.NotNil(t, err)
+			require.IsType(t, errors.NotFoundError{}, err, "error was %v", err)
+		}
+		// make sure other iterations are not touched
+		iterationsShouldPresent := []*iteration.Iteration{
+			fxt.IterationByName("root"),
+			fxt.IterationByName("child 2"),
+		}
+		for _, i := range iterationsShouldPresent {
+			_, err := rest.db.Iterations().Load(svc.Context, i.ID)
+			require.Nil(t, err)
+		}
+	})
+
+	rest.T().Run("forbidden - other user can not delete iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Identities(2, tf.SetIdentityUsernames("space owner", "other user")),
+			tf.Iterations(1))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.IdentityByUsername("other user"))
+		test.DeleteIterationForbidden(t, svc.Context, svc, ctrl, fxt.Iterations[0].ID)
+	})
+
+	rest.T().Run("success - space owner can delete iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+			if idx == 1 {
+				fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+			}
+			return nil
+		}))
+		iterationToDelete := fxt.Iterations[1]                             // non-root iteration
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0]) // get the sapce owner
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		_, err := rest.db.Iterations().Load(svc.Context, iterationToDelete.ID)
+		require.NotNil(t, err)
+		require.IsType(t, errors.NotFoundError{}, err, "error was %v", err)
+	})
+
+	rest.T().Run("unauthorized - invalid user can not delete iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.Iterations(1))
+		svc, ctrl := rest.UnSecuredController()
+		test.DeleteIterationUnauthorized(t, svc.Context, svc, ctrl, fxt.Iterations[0].ID)
+	})
+
+	rest.T().Run("success - update workitems for deleted iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+				if idx == 1 {
+					fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+				}
+				return nil
+			}),
+			tf.WorkItems(5, func(fxt *tf.TestFixture, idx int) error {
+				wi := fxt.WorkItems[idx]
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				return nil
+			}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.Iterations[1]
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		wis, err := rest.db.WorkItems().LoadByIteration(svc.Context, iterationToDelete.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+	})
+
+	// Following test creates the strucute shown in diagram
```

Comment:
minor typo: "structure"

Reply:
my bad again! fixed in https://github.com/fabric8-services/fabric8-wit/pull/1683/commits/21a922cabd1ddca24311e954ec69455efd52b824